mantidproject/mantid
# Add Phi Options to SANS Table

**Description of work.**

_Note that branch points to incorrect issue as this PR originally contained solutions to multiple issues_

Include PhiMin, PhiMax, UseMirror as parameters setable in the JobTreeView options column in SANS v2

**Report to:**  Nobody

**To test:**
1. Interfaces -> SANS -> SANS v2
2. Load any acceptable user and batch file
3. In the options column, type PhiMin=x, PhiMax=y, UseMirror=z, where x and y are floats, and z is either some truthy (e.g. True, 1, yes) or falsy (e.g. False, 0, no) statement
4. Click off the options column
5. Click the settings tab on the left hand side of the GUI, and the the `mask` tab at the top
6. Click `Display Mask`
7. The Displayed mask should show a radial section of the detector (with parts greyed out) e.g. if you typed "PhiMin=0, PhiMax=115, UseMirror=True" you should see the detector from angle 0-115 and angle 180-295
8. If you gave `UseMirror` neither a truthy or falsy statement, a value should not be taken and a readable error message should appear in the log

Fixes #23969 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
